File: 卡码/代码随想录/JS/哈希/05-四数相加/LeetCode_0454.js
//454. 四数相加 II
const fourSumCount=(nums1,nums2,nums3,nums4)=>{
  let map=new Map();
  let sum=0;
  nums1.forEach(num1=>{
    nums2.forEach(num2=>{
      let target=num1+num2;
      map.set(target,(map.get(target)||0)+1);
    });
  });
  nums3.forEach(num3=>{
    nums4.forEach(num4=>{
      let target=-(num3+num4);
      sum+=map.get(target)||0;
    });
  });
  return sum;
}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/哈希/_05_四数相加/LeetCode_0454.java
package com.kurumi.leetCode.codeThoughts.哈希._05_四数相加;

import java.util.HashMap;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 454. 四数相加 II
 */
public class LeetCode_0454 {
    static class Solution {
        public int fourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
            HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
            for (int num1 : nums1)
                for (int num2 : nums2) {
                    int target = num1 + num2;
                    if (map.containsKey(target))
                        map.put(target, map.get(target) + 1);
                    else
                        map.put(target, 1);
                }
            int sum = 0;
            for (int num3 : nums3)
                for (int num4 : nums4)
                    if (map.containsKey(-(num3 + num4)))
                        sum += map.get(-(num3 + num4));
            return sum;
        }
    }
}
